cmd/krel/cmd/release_notes.go

```diff
@@ -113,12 +141,85 @@ func runReleaseNotes() (err error) {
 	logrus.Infof("Using start tag %v", start)
 	logrus.Infof("Using end tag %v", tag)
 
-	_, err = releaseNotesFrom(start)
+	if releaseNotesOpts.sigReleaseFork == "" && releaseNotesOpts.jsonFile == "" && releaseNotesOpts.mdFile == "" {
+		logrus.Fatal("No output selected. You have to specify at least one of --md-file --json-file or --sigrelease-fork")
+	}
+
+	// TODO: Check output flags, dirs and repos before running the gatherer
+
+	result, err := releaseNotesFrom(start)
 	if err != nil {
 		return errors.Wrapf(err, "generating release notes")
 	}
 
-	//TODO: implement PR creation for k-sigs/release-notes and k/sig-release
+	// Create RN draft PR
+	if releaseNotesOpts.sigReleaseFork != "" {
+		err = createDraftPR(tag, result)
+		if err != nil {
+			return errors.Wrap(err, "Failed to create release notes draft PR")
+		}
+	}
+
+	if releaseNotesOpts.mdFile != "" {
+		err = ioutil.WriteFile(releaseNotesOpts.mdFile, []byte(result.markdown), 0644)
+		if err != nil {
+			return errors.Wrap(err, "Failed to write markdown file")
+		}
+	}
+
+	if releaseNotesOpts.jsonFile != "" {
+		err = ioutil.WriteFile(releaseNotesOpts.jsonFile, []byte(result.json), 0644)
+		if err != nil {
+			return errors.Wrap(err, "Failed to write JSON file")
+		}
+	}
+
+	// TODO: implement PR creation for k-sigs/release-notes
+	return nil
+}
+
+// createDraftPR pushes the release notes draft to the users fork
+func createDraftPR(tag string, result *releaseNotesResult) error {
+	repoparts := strings.Split(releaseNotesOpts.sigReleaseFork, "/")
+	if len(repoparts) != 2 {
+		logrus.Fatal("Repo string is not well formed, should be user/reponame")
+	}
+
+	s, err := util.TagStringToSemver(tag)
+	if err != nil {
+		return errors.Wrapf(err, "no valid tag: %v", tag)
+	}
+
+	// Clone the user's fork of k/sig-release
+	srrepo, err := git.CloneOrOpenGitHubRepo("", repoparts[0], repoparts[1], true)
+	if err != nil {
+		return errors.Wrapf(err, "cloning %s", releaseNotesOpts.sigReleaseFork)
+	}
+
+	targetdir := srrepo.Dir() + fmt.Sprintf("/releases/release-%d.%d", s.Major, s.Minor)
+	logrus.Debugf("Release notes markdown will be written to %s", targetdir)
+
+	// Ugly hack to match repo.Checkout
```

Comment:
Why is the ugly hack necessary?

Reply:
The Commit() function arguments takes a revision and arguments (Checkout(rev string, args ...string)) . I need to pass -b to git but if i send it in the arguments it appends it at the end and it breaks.